lib/ec2imgutils/ec2uploadimg.py

```diff
@@ -647,7 +647,7 @@ def _find_device_name(self, device_size):
                 continue
             this_device_size = device['size']
             unit = this_device_size[-1]
-            size = int(this_device_size[:-1])
+            size = float(this_device_size[:-1])
```

Comment:
We need to do interger math. Otherwise it could happen that our known device_size, 10 as interger, but 10.0 as float can be missed, because 10.0 * 1 could end up as 10.00000000001 or 9.99999999999. Since we do not care about ephemeral disks, catch the exception and move on.

Reply:
What about casting to an int after the multiplier? I do not think we want to truncate the raw disk size in case it is something like 10.5 as we could end up with a disk that is too small.